### ISSUE TITLE
Ingore fallthrough for consecutive case statements in `no-fallthrough`

### ISSUE THREAD

**Issue by mathieumg** [eslint/eslint#6588]
At first I was suggesting the rule below, but after playing around a bit in the REPL I figured this is already handled by default. (Unless I am mistaken) I had never activated that rule because I thought it wasn't handling that case. I'll use this issue to track improving the documentation on that matter in `no-fallthrough`. (I know issues aren't _required_ for doc, but they aren't prohibited, I imagine, to keep track of things)

**What version of ESLint are you using?**

3.0.0

**What rule do you want to change?**

`no-fallthrough`, add a `allowConsecutiveCases` option. (Name to be discussed...)

**What code should be flagged as incorrect with this change?**

Actually, the following would be marked as correct with the new option:

```js
switch (something) {
  case 1:
  case 2:
  case 3:
    // something.
    break;

  // etc.
}
```

**What happens when the rule is applied to this code now?**

Basically, the current `commentPattern` option covers the case of ensuring that fallthrough cases with side effects aren't so by accident; but I believe when case statements are consecutive it's most likely intentional/by design to handle multiple cases with the exact same code. The new option would allow consecutive case statements to be considered valid, even with the rule on and no comment specified.

**Comment by platinumazure** [comment]
Thanks for the issue!

I'm a little unclear on exactly what you're getting at. Are you saying that consecutive cases are allowed by default, but that when you specify a `commentPattern`, consecutive cases become flagged? If so, I'd say that's a bug. If not, could you please clarify your post?

**Comment by mathieumg** [comment]
I just stumbled upon https://github.com/eslint/eslint/issues/5799 and it mentions there do is an example of that use case in the doc, I just totally missed it. The rule works as I would expect, sorry for the noise.
